src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -49,48 +49,61 @@
   public static final String PYTHON = "python";
   public static final String RUBY = "ruby";
 
-  private static final String DEFAULT_SNIPPET_FILE = "discovery_fragment.snip";
+  private static final String SNIPPET_FILE_ADC = "discovery_fragment.snip";
+  private static final String SNIPPET_FILE_3LO = "discovery_fragment_3lo.snip";
+  private static final String SNIPPET_FILE_API_KEY = "discovery_fragment_api_key.snip";
```

Comment:
I wonder if we really want to put these into separate files? Looking at Victor's doc, 3LO looks a little gnarly, but API key shouldn't be too hard to fit into existing file right? I fear that the number of files will increase template duplication. I think duplication in template is a lot worse than code duplication, since a lot of tools that we use to help us maintain source code doesn't really work on templates.